alg/teca_component_area_filter.cxx

```diff
@@ -287,6 +287,18 @@ const_p_teca_dataset teca_component_area_filter::execute(
     // get the output metadata to add results to after the filter is applied
     teca_metadata &out_metadata = out_mesh->get_metadata();
 
+    long mask_value = this->mask_value;
+    if (this->mask_value < 0)
```

Comment:
see the comment in `alg/teca_2d_component_area.h` about this type of check precluding negative values for the mask value.

Reply:
yup, agree and I've changed to test to `this->mask_value == -1`